File: mobile/composeApp/src/commonMain/kotlin/io/j3mobile/protocol/ServerConfig.kt
package io.j3mobile.protocol

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

/** Server configuration snapshot. */
@Serializable
data class ServerConfig(
    val cwd: String,
    val keybindingsConfigPath: String,
    val keybindings: JsonElement? = null,
    val issues: List<ServerConfigIssue> = emptyList(),
    val providers: List<ServerProviderStatus> = emptyList(),
    val availableEditors: List<String> = emptyList(),
)

/** Describes the status of an LLM provider. */
@Serializable
data class ServerProviderStatus(
    val provider: ProviderKind,
    val status: String,
    val available: Boolean,
    val authStatus: String,
    val checkedAt: String,
    val message: String? = null,
)

/** Describes a configuration problem. */
@Serializable
data class ServerConfigIssue(
    val kind: String,
    val message: String,
    val index: Int? = null,
)

File: server/internal/protocol/server_config.go
package protocol

import "encoding/json"

// ServerConfig is the server configuration snapshot.
type ServerConfig struct {
	Cwd                   string                 `json:"cwd"`
	KeybindingsConfigPath string                 `json:"keybindingsConfigPath"`
	Keybindings           json.RawMessage        `json:"keybindings"`
	Issues                []ServerConfigIssue    `json:"issues"`
	Providers             []ServerProviderStatus `json:"providers"`
	AvailableEditors      []string               `json:"availableEditors"`
}

// ServerProviderStatus describes the status of an LLM provider.
type ServerProviderStatus struct {
	Provider   ProviderKind `json:"provider"`
	Status     string       `json:"status"`
	Available  bool         `json:"available"`
	AuthStatus string       `json:"authStatus"`
	CheckedAt  string       `json:"checkedAt"`
	Message    *string      `json:"message,omitempty"`
}

// ServerConfigIssue describes a configuration problem.
type ServerConfigIssue struct {
	Kind    string `json:"kind"`
	Message string `json:"message"`
	Index   *int   `json:"index,omitempty"`
}
